tests/gamma/csm_observability_test.py

```diff
@@ -45,39 +110,264 @@ def setUpClass(cls):
+                filter=self.build_filter_str(
+                    filter_str_template, {"metric": metric}
+                ),
+                interval=interval,
+                view=monitoring_v3.ListTimeSeriesRequest.TimeSeriesView.FULL,
+            )
+            time_series = []
+            metric_log_lines.append(f"Metric: {metric}")
+            for series in response:
+                metric_log_lines.append(" Metric Labels:")
+                for label_key, label_value in series.metric.labels.items():
+                    metric_log_lines.append(f"  {label_key}: {label_value}")
+                metric_log_lines.append(" Resource Labels:")
+                for label_key, label_value in series.resource.labels.items():
+                    metric_log_lines.append(f"  {label_key}: {label_value}")
+                time_series.append(series)
+            results[metric] = time_series
+        logger.info("\n".join(metric_log_lines))
+        return results
+
+    # A helper function to check whether at least one of the "points" whose
+    # mean should be within 5% of ref_bytes.
+    def assertAtLeastOnePointWithinRange(self, points, ref_bytes):
+        for point in points:
+            if point.value.distribution_value.mean > (
+                ref_bytes * 0.95
+            ) and point.value.distribution_value.mean < (ref_bytes * 1.05):
+                return
+        self.fail(f"No data point with ~{ref_bytes} bytes found")
+
+    # 1. actual_labels must have all the keys in expected_labels
+    #
+    # 2. actual_labels must have all the correct key: value pairs
+    #    in expected_labels unless the expected value is
+    #    VALUE_NOT_CHECKED
+    #
+    # 3. actual_labels may have extra keys that we don't care about
+    def assertExpectedLabels(self, expected_labels, actual_labels):
+        # Test whether actual_labels contains all the keys that we are
+        # expected
+        for label_key in expected_labels.keys():
+            self.assertIn(label_key, actual_labels)
+
+        # Test the label values against the expected dict
+        # Ignore those keys that were marked VALUE_NOT_CHECKED
+        self.assertDictEqual(
+            {
+                key: value
+                for key, value in expected_labels.items()
+                if value != VALUE_NOT_CHECKED
+            },
+            {
+                key: actual_labels[key]
+                for key, value in expected_labels.items()
+                if value != VALUE_NOT_CHECKED
+            },
+        )
+
     def test_csm_observability(self):
         # TODO(sergiitk): [GAMMA] Consider moving out custom gamma
         #   resource creation out of self.startTestServers()
         with self.subTest("1_run_test_server"):
-            test_server: _XdsTestServer = self.startTestServers(
-                enable_csm_observability=True
+            start_secs = int(time.time())
+            self.test_server = self.startTestServers(
```

Comment:
It must have been an artifact after several refactors. I remember having to do this because I am making a separate function and need to pass in `test_client`. But after several refactors the need isn't there anymore. Removed all those.